sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusSenderClientTest.java

```diff
@@ -0,0 +1,229 @@
+
+        //Act
+        ServiceBusMessageBatch batchMessage = sender.createBatch();
+
+        //Assert
+        Assertions.assertEquals(MAX_MESSAGE_LENGTH_BYTES, batchMessage.getMaxSizeInBytes());
+        Assertions.assertEquals(0, batchMessage.getCount());
+        verify(asyncSender, times(1)).createBatch();
+
+    }
+
+    /**
+     * Verifies we cannot create a batch if the options size is larger than the link.
+     */
+    @Test
+    void createBatchWhenSizeTooBigThanOnSendLink() {
+        // Arrange
+        int maxLinkSize = 1024;
+        int batchSize = maxLinkSize + 10;
+
+        // This event is 1024 bytes when serialized.
+        final CreateBatchOptions options = new CreateBatchOptions().setMaximumSizeInBytes(batchSize);
+        when(asyncSender.createBatch(options)).thenThrow(new IllegalArgumentException("too large size"));
+
+        // Act & Assert
+        try {
+            sender.createBatch(options);
+            Assertions.fail("Should not have created batch because batchSize is bigger than the size on SenderLink.");
+        } catch (Exception ex) {
+            Assertions.assertTrue(ex instanceof IllegalArgumentException);
+        }
+
+        verify(asyncSender, times(1)).createBatch(options);
+    }
+
+    /**
+     * Verifies that the producer can create a batch with a given {@link CreateBatchOptions#getMaximumSizeInBytes()}.
+     */
+    @Test
+    void createsMessageBatchWithSize() {
+        // Arrange
+        int maxLinkSize = 10000;
+        int batchSize = 1024;
+
+        // Overhead when serializing an event, to figure out what the maximum size we can use for an event payload.
+        int eventOverhead = 46;
+        int maxEventPayload = batchSize - eventOverhead;
+
+        // This is 1024 bytes when serialized.
+        final ServiceBusMessage message = new ServiceBusMessage(new byte[maxEventPayload]);
+
+        final ServiceBusMessage tooLargeMessage = new ServiceBusMessage(new byte[maxEventPayload + 1]);
+        final CreateBatchOptions options = new CreateBatchOptions().setMaximumSizeInBytes(batchSize);
+        ServiceBusMessageBatch batch =  new ServiceBusMessageBatch(batchSize, null, tracerProvider,
+            messageSerializer);
+        when(asyncSender.createBatch(options)).thenReturn(Mono.just(batch));
+
+        // Act & Assert
+        ServiceBusMessageBatch messageBatch = sender.createBatch(options);
+
+        Assertions.assertEquals(batchSize, messageBatch.getMaxSizeInBytes());
+        Assertions.assertTrue(messageBatch.tryAdd(message));
+        Assertions.assertFalse(messageBatch.tryAdd(tooLargeMessage));
+
+    }
+
+    /**
+     * Verifies that sending multiple message will result in calling sender.send(MessageBatch).
+     */
+    @Test
+    void sendMultipleMessages() {
+        // Arrange
+        final int count = 4;
+        final byte[] contents = TEST_CONTENTS.getBytes(UTF_8);
+        final ServiceBusMessageBatch batch = new ServiceBusMessageBatch(256 * 1024,
```

Comment:
You don't need this. You just need to assert that the send(ServiceBusMessage) is called on the asyncClient. And same with the one below.